tests/testthat/test-nest.R

```diff
@@ -233,15 +244,29 @@ test_that("can unnest empty data frame", {
   expect_equal(out, tibble(x = integer(), y = unspecified()))
 })
 
+test_that("unnesting bare lists of NULLs is equivalent to unnesting empty lists", {
+  skip("until rebased on #1140, which fixes `unchop()` with `list(NULL)` cols")
+  df <- tibble(x = 1L, y = list(NULL))
+  out <- unnest(df, y)
+  expect_identical(out, tibble(x = integer(), y = unspecified()))
+})
+
 test_that("unnest() preserves ptype", {
   tbl <- tibble(x = integer(), y = list_of(ptype = tibble(a = integer())))
   res <- unnest(tbl, y)
   expect_equal(res, tibble(x = integer(), a = integer()))
 })
 
-test_that("errors on bad inputs", {
+test_that("unnesting typed lists of NULLs retains ptype", {
+  skip("until rebased on #1140, which fixes `unchop()` with `list_of(NULL)` cols")
+  df <- tibble(x = 1L, y = list_of(NULL, .ptype = tibble(a = integer())))
+  out <- unnest(df, y)
+  expect_identical(out, tibble(x = integer(), a = integer()))
+})
+
+test_that("skips over vector columns", {
   df <- tibble(x = integer(), y = list())
-  expect_error(unnest(df, x), "list of vectors")
+  expect_identical(unnest(df, x), df)
```

Comment:
This one used to be an error, but only in the size 0 case? i.e. on master:

``` r
library(tidyr)

df <- tibble(x = 1L, y = 1)
unnest(df, y)
#> # A tibble: 1 x 2
#>       x     y
#>   <int> <dbl>
#> 1     1     1

df <- tibble(x = integer(), y = double())
unnest(df, y)
#> Error: Input must be list of vectors
```

`unnest()` is generally pretty forgiving about trying to unnest a non list-col (which pretty much results in a no-op), so I think this should work.